### DISCUSSION TITLE
Add metadata to images in bucket

### DISCUSSION THREAD

**Discussion by LavaGolem** [supabase/supabase#13900] [Questions]
Hi, I would like to keep some custom metadata like tags, titles, etc. Together with images (files) in the buckets. 

Is something like that possible? And how to do it using SDK. 

Thanks

**Comment by GaryAustin1** [comment]
You would need to create your own table and link based on pathname or storage.object id column.  
There is no API feature to do so.

**Comment by neerax** [comment]
Supabase storage claims to be AWS S3 compatible, so it should support x-amz-meta-x headers to set custom metadata properties and clients library should handle that?

**Reply by GaryAustin1** [reply]
Not quite sure what you mean by AWS S3 compatible.   Supabase Storage-API uses AWS S3 storage on their hosted instances, but you do not have access to it directly and the API stores file info in a storage.objects table with a metadata column they (SB) created.   You have no access to S3 directly.

**Comment by neerax** [comment]
Hi Gary, thanks for the fast reply.
On https://github.com/supabase/storage-api the about section says: "S3 compatible object storage service that stores metadata in Postgres".
The term "compatible" may lead to confusion, as other object services use it to claims S3 "API" compatibility, e.g. MinIO declares on its' home page: "MinIO is a high-performance, S3 compatible object store."

Regards

**Reply by GaryAustin1** [reply]
Yeah, the description in the same place is more accurate.  And it for sure does not support any s3 compatible calls or let you use the current s3 storage behind the scenes directly.

A scalable, light-weight object storage service.

    Uses Postgres as it's datastore for storing metadata
    Authorization rules are written as Postgres Row Level Security policies
    Integrates with S3 as the storage backend (with more in the pipeline!)
    Extremely lightweight and performant
